### PR TITLE
Feature/1415

### PR DESCRIPTION
**Pull request by Elaine-Krauss-TCG**
[App#1414](https://app.zenhub.com/workspaces/fecfile-online-619e578e68408b001c831251/issues/gh/fecgov/fecfile-web-app/1414)
[App#1415](https://app.zenhub.com/workspaces/fecfile-online-619e578e68408b001c831251/issues/gh/fecgov/fecfile-web-app/1415)

[App PR](https://github.com/fecgov/fecfile-web-app/pull/1684)
[API PR](https://github.com/fecgov/fecfile-web-api/pull/685)

### CODE REVIEW

File: front-end/src/app/shared/components/inputs/additional-info-input/additional-info-input.component.ts [hunk 1]
```diff
@@ -23,8 +23,8 @@ export class AdditionalInfoInputComponent extends BaseInputComponent implements
       );
     }
 
-    if (this.transaction?.transactionType?.memoTextPrefix) {
-      this.initPrefix(this.templateMap.text4000, this.transaction?.transactionType?.memoTextPrefix);
```
Comment (mjtravers) [inline front-end/src/app/shared/components/inputs/additional-info-input/additional-info-input.component.ts:27]: If we lose memoTextPrefix, the MULTISTATE_INDEPENDENT_EXPENDITURE will have its transaction type setting lost. It looks like the text_prefix will need to account for a default value.

See here:
https://github.com/fecgov/fecfile-web-app/blob/develop/front-end/src/app/shared/models/transaction-type.model.ts#L85
https://github.com/fecgov/fecfile-web-app/blob/develop/front-end/src/app/shared/models/transaction-types/MULTISTATE_INDEPENDENT_EXPENDITURE.model.ts#L46-L47

File: front-end/src/app/shared/components/transaction-type-base/transaction-memo.utils.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ export class TransactionMemoUtils {
       if (text && text.length > 0) {
         const memo_text = MemoText.fromJSON({
           text4000: text,
+          text_prefix: transaction.memo_text?.text_prefix,
```
Comment (mjtravers) [inline front-end/src/app/shared/components/transaction-type-base/transaction-memo.utils.ts]: Would it make sense to refactor the two static functions in in this file and put them in the additional-info-input component? The transaction and form are available in the component and then memo HTML and component code are self-contained. Then we could eliminate this file:

https://github.com/fecgov/fecfile-web-app/blob/develop/front-end/src/app/shared/components/transaction-type-base/transaction-memo.utils.ts

File: front-end/src/app/shared/utils/reatt-redes/redesignated.utils.ts [hunk 3]
```diff
@@ -1,9 +1,23 @@
...
+        if (transaction.memo_text) {
+          transaction.memo_text.text_prefix = prefix;
+          transaction.memo_text.text4000 = prefix + transaction?.memo_text?.text4000;
+        } else {
+          transaction.memo_text = MemoText.fromJSON({
+            text_prefix: prefix,
+            text4000: prefix,
+          });
+        }
+      }
```
Comment (mjtravers) [inline front-end/src/app/shared/utils/reatt-redes/redesignated.utils.ts]: This block of code is very similar to the comparable chunk in rattributted.utils.ts. Could we consolidate the logic into a static function in reatt-redes.utils.ts? The new function would get passed the transaction and either the flavor of purpose_descrip it needed.

**Review by mjtravers — CHANGES_REQUESTED** [review]

File: front-end/src/app/shared/utils/reatt-redes/redesignated.utils.ts (hunk 3 above)
Comment (Elaine-Krauss-TCG) [reply front-end/src/app/shared/utils/reatt-redes/redesignated.utils.ts]: Done!

File: front-end/src/app/shared/components/inputs/additional-info-input/additional-info-input.component.ts (hunk 1 above)
Comment (Elaine-Krauss-TCG) [reply front-end/src/app/shared/components/inputs/additional-info-input/additional-info-input.component.ts:27]: Ah, that's a really good catch!  I've fixed it, and I've tested that Multistate Independent Expenditures work with the fix.

File: front-end/src/app/shared/components/transaction-type-base/transaction-memo.utils.ts (hunk 2 above)
Comment (Elaine-Krauss-TCG) [reply front-end/src/app/shared/components/transaction-type-base/transaction-memo.utils.ts]: They're not used in the additional info component, but I've moved them to the one place where they are used, the transaction form utils.

**Review by mjtravers — APPROVED** [review]

**Pull request merged**